jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
> You said that should be equal to the id, which id did you mean? the transactionId?

Yes, in theory, that should be the behavior (whenever an object has both a key and id fields, if the key isn't specified, the system should set it to the auto-generated id).

> I debugged for quite a while until I got [this!](https://github.com/javier-gomez-hs/killbill/blob/b31a8ad2e463c5bf9869b4d2c69078e3ea5ecb55/payment/src/main/java/org/killbill/billing/payment/core/sm/control/DefaultControlInitiated.java#L94-L94)

Ah yes, that's exactly the issue https://github.com/killbill/killbill/issues/349. 

Is it possible to remove `stateContext.setPaymentTransactionExternalKey` from `DefaultControlInitiated`? The idea would be that the system will set it automatically when the model is built (see https://github.com/killbill/killbill/blob/master/payment/src/main/java/org/killbill/billing/payment/dao/PaymentTransactionModelDao.java#L60 for instance). I'm not quite sure why it's different in that case. It's possible that it's because it's needed in `InvoicePaymentControlPluginApi` (in the `priorCall`), which is before the model is built.

Could you give it a try if it's easy to remove it? If it breaks all sorts of tests, don't worry about the check. Let's still pass `null` from JAX-RS and we can tackle the inconsistency in a separate PR.

Thanks!